### PR TITLE
Custom Address HRPs

### PR DESCRIPTION
**Pull request by michavie**
Added support for custom Address HRPs + started adopting the new SDK specs

### CODE REVIEW

File: src/Address.php [hunk 1]
```diff
@@ -7,47 +7,55 @@
...
         $res = convertBits($decoded, count($decoded), 5, 8, false);
         $pieces = array_map(fn ($bits) => dechex($bits), $res);
         $hex = array_reduce($pieces, fn ($carry, $hex) => $carry . str_pad($hex, 2, "0", STR_PAD_LEFT));
 
-        return new Address($hex);
+        return new Address($hex, $hrp);
     }
 
-    public static function fromBase64(string $value): Address
+    public static function newFromBase64(string $value, string $hrp = self::DEFAULT_HRP): Address
```
Comment (andreibancioiu) [inline src/Address.php:51]: Good point, maybe we should add this in specs. Useful when handling (at a lower level) topics of some events etc.

**Review by andreibancioiu — APPROVED** [review]

File: src/Address.php (hunk 1 above)
Comment (razvanphp) [reply src/Address.php:51]: but.. why do we need to rename the function name, if the second added param has a default? It would still be BC... right?

Comment (michavie) [reply src/Address.php:51]: @razvanphp The rename is part of aligning with the new sdk specs. Note that we did not remove the old ones like `fromBase64` but only deprecated them with a reference to the new functions

Comment (razvanphp) [reply src/Address.php:51]: got it, sorry, did not check the new specs. Maybe we should link them somewhere.

**Review by razvanphp — APPROVED** [review]

**Pull request merged**